File: sortingAlgorithms/bubbleSort/bubble_sort.rb
def bubble_sort(arr)
    for i in (arr.size - 1).downto(0) do
        no_swaps = true
        i.times do |j|
            if arr[j] > arr[j+1]
                arr[j], arr[j+1] = arr[j+1], arr[j]
                no_swaps = false
            end
        end
        break if no_swaps
    end
    return arr
end



puts bubble_sort([1,2,5,32,9,6,3])

File: sortingAlgorithms/bubbleSort/bubble_sort.py
def bubble_sort(arr):
    for i in range(len(arr) - 1, -1, -1):
        no_swaps = True
        for j in range(0, i):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                no_swaps = False
        if no_swaps: break
    return arr


print(bubble_sort([1,2,5,32,9,6,3]))